src/animated/Controller.ts

```diff
@@ -241,19 +242,30 @@ class Controller<DS extends object = any> {
   }
 
   private _run(guid: number, props: UpdateProps<DS>, onEnd?: OnEnd) {
-    // Never call `onStart` for immediate animations
-    if (!props.immediate) {
-      const { onStart } = props
-      // Allow `useCallback` to prevent multiple calls
-      if (onStart && onStart !== this.props.onStart) onStart()
-    }
     if (is.arr(props.to) || is.fun(props.to)) {
       this._runAsync(guid, props, onEnd)
     } else {
       this._diff(props)._start(onEnd)
     }
   }
 
+  private _willStart(props: SpringProps<any>, oldProps: SpringProps<any>) {
```

Comment:
Note: The `props` argument is only the "update" props. 
Not the new value of `this.props`.